crates/pallets/starknet/src/lib.rs

```diff
@@ -1166,6 +1107,77 @@ impl<T: Config> Pallet<T> {
+
+        let execution_results = execute_txs_and_rollback::<T>(
+            &transactions,
+            &Self::get_block_context(),
+            chain_id,
+            false,
+            disable_validation,
+            disable_fee_charge,
+        );
+
+        let get_function_invocation =
+            |call_info: Option<&CallInfo>| -> Result<Option<FunctionInvocation>, DispatchError> {
+                match call_info {
+                    Some(call_info) => Ok(Some(convert_call_info_to_function_invocation::<T>(call_info)?)),
+                    None => Ok(None),
+                }
+            };
+
+        let mut results = vec![];
+        for (tx, res) in transactions.iter().zip(execution_results.iter()) {
+            match res {
+                Ok(tx_exec_info) => {
+                    let validate_invocation = get_function_invocation(tx_exec_info.validate_call_info.as_ref())?;
+                    let fee_transfer_invocation =
+                        get_function_invocation(tx_exec_info.fee_transfer_call_info.as_ref())?;
+                    let transaction_trace = match tx {
+                        UserTransaction::Invoke(tx) => TransactionTrace::Invoke(InvokeTransactionTrace {
+                            validate_invocation,
+                            execute_invocation: convert_call_info_to_execute_invocation::<T>(
+                                tx_exec_info.execute_call_info.as_ref().ok_or(Error::<T>::MissingCallInfo)?,
+                                tx_exec_info.revert_error.as_ref(),
+                            )?,
+                            fee_transfer_invocation,
+                        }),
+                        UserTransaction::Declare(tx, _) => TransactionTrace::Declare(DeclareTransactionTrace {
+                            validate_invocation,
+                            fee_transfer_invocation,
+                        }),
+                        UserTransaction::DeployAccount(tx) => {
+                            TransactionTrace::DeployAccount(DeployAccountTransactionTrace {
+                                validate_invocation,
+                                constructor_invocation: convert_call_info_to_function_invocation::<T>(
+                                    tx_exec_info.execute_call_info.as_ref().ok_or(Error::<T>::MissingCallInfo)?,
+                                )?,
+                                fee_transfer_invocation,
+                            })
+                        }
+                    };
+                    log::info!("Simulated transaction: {:?}", transaction_trace);
+                    results.push(SimulatedTransaction {
+                        transaction_trace,
+                        fee_estimation: FeeEstimate { gas_consumed: 0, gas_price: 0, overall_fee: 0 },
+                    })
+                }
+                Err(e) => {
+                    log!(info, "Failed to simulate transactions: {:?}", e);
```

Comment:
Log error

Reply:
Maybe add a bit more info: can we log the tx for which simulation failed?

Reply:
added